nvidia_nims/vista_3d_remote_nim.ipynb

```diff
@@ -0,0 +1,523 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #55.                    return itk.imread(filepath, pixel_type=itk.SS)
</pre><p>if <code>imread</code> can read from a data stream you can use <code>z.open</code> to open streams to files in the zip file without extracting it, for large zips this can be faster.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1898/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Reply:
Streaming from zips is already done by ITK (i.e., imread) depending on the file format (e.g., you can stream MetaIO files).   It also has a very fast way of parsing directories of DICOM objects to load 3D volumes (by default loading the first series in the directory, if a series ID isn't specified).

I am not certain if ITK provides a way of specifying a collection of DICOM via gzip - I know we have written programs on top of ITK that do that, but interesting idea to add that capability to ITK's imread directly.  As you note, it would be easy.    It seems outside the scope of this notebook, or am I missing something?